Add support for adding attributes to enum variants

Kriskras99 opened the pull request:
When dealing with `serde` and `serde_as` it can be useful to add attributes to enum variants:
```rust

#[superstruct(
    variants(V2122, V20, V19, V18),
    variant_attributes(derive(Debug, Serialize, Deserialize, Clone)),
    enum_variant_attributes(serde(borrow))
)]
#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(untagged)]
pub struct ConfigFilesPath<'a> {
    #[superstruct(only(V20, V2122))]
    #[serde(borrow)]
    pub gachacontent: HipStr<'a>,
    #[superstruct(only(V20, V2122))]
    #[serde(borrow)]
    pub ftuesteps: HipStr<'a>,
    #[superstruct(only(V20))]
    #[serde(borrow)]
    pub anthology: HipStr<'a>,
    #[superstruct(only(V20, V2122))]
    #[serde(borrow)]
    pub objectives: HipStr<'a>,
    #[superstruct(only(V19, V20, V2122))]
    #[serde(borrow)]
    pub playlist: HipStr<'a>,
    #[superstruct(only(V20, V2122))]
    #[serde(borrow)]
    pub portraitborders: HipStr<'a>,
    #[superstruct(only(V2122))]
    #[serde(borrow)]
    pub quickplayrules: HipStr<'a>,
    #[superstruct(only(V20, V2122))]
    #[serde(borrow)]
    pub scheduledquests: HipStr<'a>,
    #[superstruct(only(V18, V19))]
    #[serde(borrow)]
    pub dmconfig: HipStr<'a>,
    #[superstruct(only(V19))]
    #[serde(borrow)]
    pub postcards: HipStr<'a>,
}
```

This PR does need tests, but I'm not sure how to tests this. Do you have any ideas for testing this?

michaelsproul commented:
Sorry this slipped off my radar

To test this I think your example using `serde(borrow)` could be a good start. We could write a test that only passes (or only compiles) if `serde(borrow)` is being applied.

Kriskras99 commented:
Also fell of my radar, I was just using a git dependency in my project.

I've added a test that fails to compile when the enum_variant_attributes line is removed.

Kriskras99 commented:
I've now also added documentation for `specific_enum_variant_attributes` and fixed some small issues in the book. It's ready for review now @michaelsproul

Kriskras99 commented:
Friendly ping, are you still interested in merging this?

michaelsproul commented:
Oh yes sorry, fell off my radar but I can jump onto this next week 🙏

Kriskras99 commented on tests/basic.rs:
```diff
@@ -152,3 +153,23 @@ fn no_getter() {
 }
+
+#[test]
+#[allow(dead_code)]
```
I've added a test for `rename`. The `serde(borrow)` is already being tested, as it would be a compile error otherwise (try removing any of the borrow statements).

michaelsproul reviewed (approved):
Thanks!

Kriskras99 commented:
Nice! Thank you so much!
